x/ibc/core/23-commitment/types/merkle.go

```diff
@@ -68,51 +68,53 @@ func (mp MerklePrefix) Empty() bool {
 var _ exported.Path = (*MerklePath)(nil)
 
 // NewMerklePath creates a new MerklePath instance
-func NewMerklePath(keyPathStr []string) MerklePath {
-	merkleKeyPath := KeyPath{}
-	for _, keyStr := range keyPathStr {
-		merkleKeyPath = merkleKeyPath.AppendKey([]byte(keyStr), URL)
-	}
-
+// The keys must be passed in from root-to-leaf order
+func NewMerklePath(keyPath ...string) MerklePath {
+	kp := keyPath[:]
 	return MerklePath{
-		KeyPath: merkleKeyPath,
+		KeyPath: kp,
 	}
 }
 
 // String implements fmt.Stringer.
 func (mp MerklePath) String() string {
-	return mp.KeyPath.String()
+	return "/" + strings.Join(mp.KeyPath, "/")
```

Comment:
q: does it need to start with / ?

Reply:
what if the key paths are empty, will it return `"//"`? I'd like to see a few test cases if possible

Reply:
Some godoc on why it does this as well

Reply:
Follows tendermint KeyPath implementation. Documented